staging/src/k8s.io/kubectl/pkg/cmd/create/create_namespace.go

```diff
@@ -36,16 +49,42 @@ var (
 	  kubectl create namespace my-namespace`))
 )
 
-// NamespaceOpts is the options for 'create namespace' sub command
-type NamespaceOpts struct {
-	CreateSubcommandOptions *CreateSubcommandOptions
+// NamespaceOptions is the options for 'create namespace' sub command
+type NamespaceOptions struct {
+	// PrintFlags holds options necessary for obtaining a printer
+	PrintFlags *genericclioptions.PrintFlags
+	// Name of resource being created
+	Name string
+	// StructuredGenerator is the resource generator for the object being created
+	StructuredGenerator generate.StructuredGenerator
+	DryRunStrategy      cmdutil.DryRunStrategy
+	DryRunVerifier      *resource.DryRunVerifier
+	CreateAnnotation    bool
+	FieldManager        string
+
+	Namespace        string
+	EnforceNamespace bool
+
+	Mapper        meta.RESTMapper
+	DynamicClient dynamic.Interface
+
+	PrintObj func(obj kruntime.Object, out io.Writer) error
+
+	genericclioptions.IOStreams
+}
+
+// NewNamespaceOptions creates a new *NamespaceOptions with sane defaults
+func NewNamespaceOptions(ioStreams genericclioptions.IOStreams) *NamespaceOptions {
+	return &NamespaceOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("created").WithTypeSetter(scheme.Scheme),
+		IOStreams:  ioStreams,
+	}
 }
 
 // NewCmdCreateNamespace is a macro command to create a new namespace
 func NewCmdCreateNamespace(f cmdutil.Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
-	options := &NamespaceOpts{
-		CreateSubcommandOptions: NewCreateSubcommandOptions(ioStreams),
-	}
+
+	options := NewNamespaceOptions(ioStreams)
```

Comment:
small nit, but for better reading and comparison with other create cmd, please rename this var to o (instead of options) and call it this way on options.Complete (o.Complete) and so on :)